pkg/healthchecker/health_checker.go

```diff
@@ -82,7 +82,8 @@ func getUptimeFunc(service string) func() (time.Duration, error) {
 		if len(val) < 2 {
 			return time.Duration(0), errors.New("could not parse the service uptime time correctly")
 		}
-		t, err := time.Parse(types.UptimeTimeLayout, val[1])
+		timeEndIndex := strings.LastIndex(val[1], " ")
+		t, err := time.Parse(types.UptimeTimeLayout, val[1][:timeEndIndex])
```

Comment:
IIUC, if you run NPD in a container, it may not be in the same time zone with the host, in that case would this cause any problem?

What is the problem if we don't make this change? I don't quite get it after reading the PR description.

Reply:
The resoult for the cmd out is like this:
```$ systemctl show kubelet --property=InactiveExitTimestamp
InactiveExitTimestamp=Wed 2021-04-14 01:11:39 -07
```
But types.UptimeTimeLayout is `UptimeTimeLayout          = "Mon 2006-01-02 15:04:05"`
It is failed to parse time

Reply:
The `UptimeTimeLayout` was `"Mon 2006-01-02 15:04:05 UTC"` before your change.

And in your PR description, the systemctl output was:
```
[root@kubernetes-master-505-pkshx-4589367 software]# systemctl show kubelet --property=InactiveExitTimestamp InactiveExitTimestamp=Wed 2021-03-10 06:05:51 PST
```

And on my node it is also:
```
$ systemctl show docker --property=InactiveExitTimestamp
InactiveExitTimestamp=Thu 2021-04-15 17:20:37 PDT
```

Why would the parsing fail?

Reply:
I rollback to old version and retry on my node, it is still wrong and reture error: 
```
I0425 22:54:26.604844   18356 plugin.go:87] Add check result {Rule:0xc4200bd9d0 ExitStatus:2 Message:systemctl show: Sun 2021-04-25 11:22:50 PDT
error checking kubelet health: pars}
```
Then I code a test case like this:
```package main

import (
    "fmt"
    "time"
)

const UptimeTimeLayout = "Mon 2006-01-02 15:04:05 UTC"

func main() {
    strs := []string{
        "Wed 2021-03-10 03:39:13 PST",
        "Sun 2021-04-25 11:22:50 PDT",
    }

    t, err := time.Parse(UptimeTimeLayout, strs[0])
    if err != nil {
        fmt.Println("error: ", err)
    }
    t, err = time.Parse(UptimeTimeLayout, strs[0])
    if err != nil {
        fmt.Println("error: ", err)
    }
    fmt.Println(t.String())
}
```
It is also wrong:
```error:  parsing time "Wed 2021-03-10 03:39:13 PST" as "Mon 2006-01-02 15:04:05 UTC": cannot parse "PST" as " UTC"
error:  parsing time "Wed 2021-03-10 03:39:13 PST" as "Mon 2006-01-02 15:04:05 UTC": cannot parse "PST" as " UTC"
0001-01-01 00:00:00 +0000 UTC
Exiting.
```

Reply:
I see. In that case, we probably need a better fix.

Reply:
```
package main
import (
    "fmt"
    "time"
)

const UptimeTimeLayout = "Mon 2006-01-02 15:04:05 MST"

func main() {
    strs := []string{
        "Wed 2021-03-10 03:39:13 PST",
        "Sun 2021-04-25 11:22:50 PDT",
    }

    t, err := time.Parse(UptimeTimeLayout, strs[0])
    if err != nil {
        fmt.Println("error: ", err)
    }
    fmt.Println(t.String())
    t, err = time.Parse(UptimeTimeLayout, strs[1])
    if err != nil {
        fmt.Println("error: ", err)
    }
    fmt.Println(t.String())
}
```

Can you change it to MST?
> When parsing a time with a zone abbreviation like MST, if the zone abbreviation has a defined offset in the current location, then that offset is used.